assets/src/domain/eventEditor/ui/tickets/hooks/useTicketsActionMenuHandler.tsx

```diff
@@ -5,38 +5,30 @@ import EditTicketButton from '../ticketsList/actionsMenu/EditTicketButton';
 import DeleteTicketButton from '../ticketsList/actionsMenu/DeleteTicketButton';
 import AssignDatesButton from '../ticketsList/actionsMenu/AssignDatesButton';
 import TicketPriceCalculatorButton from '../ticketPriceCalculator/buttons/TicketPriceCalculatorButton';
-import { EntitySubscriptionCallback } from '@appLayout/entityActionsMenu';
+import { EntityActionsSubscriptionCb } from '@appLayout/entityActionsMenu';
 import { Ticket } from '@edtrServices/apollo/types';
-import { AdditionalTicketMenuOptions } from '../types';
 import { useStatus, TypeName } from '@appServices/apollo/status';
 
-type TicketsSubscriptionCallback = EntitySubscriptionCallback<Ticket, AdditionalTicketMenuOptions>;
+type TicketsSubscriptionCallback = EntityActionsSubscriptionCb<Ticket, 'ticket'>;
 
 const useTicketsActionMenuHandler = (): TicketsSubscriptionCallback => {
-	return useCallback<TicketsSubscriptionCallback>(
-		({ entity: ticket }, { registerMenuItem }, { ticketMenuItemProps: menuItemProps }) => {
-			registerMenuItem('editTicket', () => (
-				<EditTicketButton key={ticket.id + 'editTicket'} {...menuItemProps} />
-			));
+	return useCallback<TicketsSubscriptionCallback>(({ entity: ticket }, { registerMenuItem }) => {
+		registerMenuItem('editTicket', () => <EditTicketButton key={ticket.id + 'editTicket'} />);
 
-			registerMenuItem('assignDates', () => <AssignDatesButton id={ticket.id} {...menuItemProps} />);
+		registerMenuItem('assignDates', () => <AssignDatesButton id={ticket.id} />);
 
-			registerMenuItem('ticketPriceCalculator', () => {
-				const { isLoaded } = useStatus();
-				/* Hide price calculator unless prices are loaded */
-				return (
-					isLoaded(TypeName.prices) && <TicketPriceCalculatorButton ticketId={ticket.id} {...menuItemProps} />
-				);
-			});
+		registerMenuItem('ticketPriceCalculator', () => {
+			const { isLoaded } = useStatus();
+			/* Hide price calculator unless prices are loaded */
+			return isLoaded(TypeName.prices) && <TicketPriceCalculatorButton ticketId={ticket.id} />;
+		});
 
-			registerMenuItem('deleteTicket', () => {
-				const { isLoaded } = useStatus();
-				/* Delete button should be hidden to avoid relational inconsistencies */
-				return isLoaded(TypeName.prices) && <DeleteTicketButton id={ticket.id} {...menuItemProps} />;
-			});
-		},
-		[]
-	);
+		registerMenuItem('deleteTicket', () => {
+			const { isLoaded } = useStatus();
+			/* Delete button should be hidden to avoid relational inconsistencies */
+			return isLoaded(TypeName.prices) && <DeleteTicketButton id={ticket.id} />;
+		});
+	}, []);
```

Comment:
not saying you HAVE to do it this way, but if it were possible to wrap the isLoaded checks around the calls to `registerMenuItem()`, then you could use the same instance of `isLoaded` instead of repeatedly calling `useStatus()`. I don't know if that would even work, or have other complications, but just throwing out another idea.

Reply:
Same as above. One solution that we could try is HOC for the second argument of `registerMenuItem`

Reply:
Wrapped them inside `withIsLoaded` HOC

Reply:
ya it might make sense to have a generic `withIsLoaded` HOC that we can wrap components with to bypass their loading based on the status of an entity. So instead of having those checks here, we add them directly to the components in question, like `TicketPriceCalculatorButton`, `DeleteTicketButton`, etc